src/rpdk/core/contract/type_configuration.py

```diff
@@ -6,48 +6,50 @@
 
 LOG = logging.getLogger(__name__)
 
-TYPE_CONFIGURATION_FILE_PATH = "~/.cfn-cli/typeConfiguration.json"
-
 
 class TypeConfiguration:
     TYPE_CONFIGURATION = None
 
     @staticmethod
-    def get_type_configuration():
+    def get_type_configuration(typeconfigloc):
+        if typeconfigloc:
+            type_config_file_path = typeconfigloc
+        else:
+            type_config_file_path = "~/.cfn-cli/typeConfiguration.json"
+
         LOG.debug(
-            "Loading type configuration setting file at '~/.cfn-cli/typeConfiguration.json'"
+            "Loading type configuration setting file at %s",
+            type_config_file_path,
         )
         if TypeConfiguration.TYPE_CONFIGURATION is None:
             try:
                 with open(
-                    os.path.expanduser(TYPE_CONFIGURATION_FILE_PATH), encoding="utf-8"
+                    os.path.expanduser(type_config_file_path), encoding="utf-8"
```

Comment:
Maybe this addresses my previous comment.

Reply:
That is correct.  The path.expanduser function works on posix and windows.